LaTeX/TeX.sublime-syntax

```diff
@@ -277,30 +272,51 @@ contexts:
     - include: paragraph-pop
 
   def-definition:
-    - match: \{
-      scope: punctuation.definition.group.brace.begin.tex
-      set: def-definition-body
+    - match: (?=\{)
+      set:
+        - def-scope-as-body
+        - macro-braces-begin
     - include: paragraph-pop
     - include: else-pop
 
-  def-definition-body:
-    - meta_scope: meta.function.body.tex meta.group.brace.tex
-    - include: macro-brace-group-body
+  def-scope-as-body:
+    - meta_scope: meta.function.body.tex
+    - include: immediately-pop
 
-  # within macros, it is possible that only part of some nested struture
-  # is present. To prevent this from causing problems, here we only match
-  # elements that are simple, i.e. that do not push to the stack.
   macro-braces:
+    - match: (?=\{)
+      push: macro-braces-begin
```

Comment:
It's finally probably even better to directly consume the `{` and push into content to avoid the additional context switch.

```suggestion
    - match: \{
      scope: punctuation.definition.group.brace.begin.tex
      push: macro-braces-body
```